### ISSUE TITLE
Alpha: await algoliaClient.search doesnt accept generic anymore

### ISSUE THREAD

**Issue by loick** [algolia/api-clients-automation#2137]
Hi, 

I'm using the 5.0.0-alpha.81 version, and I used to have something like this in order to type the results:

```js
const { results } = await algoliaClient.search<MyCustomDocumentType>({ ... })
```

However, ever since the alpha.82 I believe, it's not accepting any generics anymore. Any reason for this? Maybe there is a new & much better way to do it :) 

Thanks!

**Comment by shortcuts** [comment]
Hey @loick, thanks for testing the Alpha client :D 

> However, ever since the alpha.82 I believe, it's not accepting any generics anymore. Any reason for this? Maybe there is a new & much better way to do it :)

Looking at our releases, this was changed in https://github.com/algolia/api-clients-automation/pull/1774 which was released in a .7x or something I believe.

TLDR this is due to the API endpoint allowing to search for either `hits` or `facets`, meaning the generic becomes a bit cumbersome to deal with in term of implementation, since it can be mixed.

We will work on providing helpers method named `searchForHits` and `searchForFacets` which will allow generic and ease the typing :)

**Comment by loick** [comment]
Okay, thanks for the quick answer! 🙏 Do you know when it will move to the beta stage also btw?

**Comment by shortcuts** [comment]
Hey @loick, we will provide some helpers in the next release https://github.com/algolia/api-clients-automation/pull/2138

> Okay, thanks for the quick answer! 🙏 Do you know when it will move to the beta stage also btw?

If everything goes well we will go stable next quarter!

**Comment by loick** [comment]
Thanks a lot!

**Comment by loick** [comment]
@shortcuts 
Is it possible to release the alpha90 that seem to exist in tags but not in the npm package. Thanks!

**Comment by shortcuts** [comment]
hey, the tag takes the highest version of all of the packages, you should find the new helpers method latest algoliasearch version (alpha.89)
